src/main/java/io/vertx/mqtt/impl/MqttEndpointImpl.java

```diff
@@ -653,6 +655,32 @@ public void close() {
     }
   }
 
+  public SocketAddress localAddress() {
+    synchronized (this.conn) {
+      return conn.localAddress();
```

Comment:
Hmm... I believe there was similar code without it. Now looking again it seems those codes were protocol message handlers and all commanding codes seem to have it. Adding.